app/api/chemotion/search_api.rb

```diff
@@ -97,10 +97,13 @@ def sample_structure_search(c_id = @c_id, not_permitted = @dl_s && @dl_s < 1)
 
         # TODO: implement this: http://pubs.acs.org/doi/abs/10.1021/ci600358f
         scope =
-          if params[:selection][:search_type] == 'similar'
+          case params[:selection][:search_type]
+          when 'similar'
             Sample.by_collection_id(c_id).search_by_fingerprint_sim(molfile, threshold)
-          else
+          when 'sub'
             Sample.by_collection_id(c_id).search_by_fingerprint_sub(molfile)
+          when 'subRDKit'
+            Sample.by_collection_id(c_id).search_by_rdkit_sub(molfile)
           end
```

Comment:
We discussed adding a fallback / default case for invalid `search_type`.

Reply:
The search type has a whitelist: "optional :search_type, type: String, values: %w[similar sub subRDKit]"
Do we still need a fallback / default case?